tests/k8st/create_kind_cluster.sh

```diff
@@ -0,0 +1,148 @@
+            - name: IP6
+              value: "autodetect"
+            - name: CALICO_IPV6POOL_CIDR
+              value: "fd00:10:244::/64"
+            - name: CALICO_IPV6POOL_IPIP
+              value: "Never"
+            - name: CALICO_IPV6POOL_NAT_OUTGOING
+              value: "false"
+EOF
+  # update FELIX_IPV6SUPPORT=true
+  sed -i '/FELIX_IPV6SUPPORT/!b;n;c\              value: "true"' "${yaml}"
+}
+
+echo "kubernetes dualstack requires ipvs mode kube-proxy for the moment."
+MODULES=("ip_vs" "ip_vs_rr" "ip_vs_wrr" "ip_vs_sh" "nf_conntrack_ipv4")
+for m in "${MODULES[@]}"; do
+  checkModule $m
+  if [[ $? -eq 1 ]]; then
+    echo "Could not find kernel module $m. install it..."
+    sudo modprobe $m
+  fi
+done
+echo
+
+echo "Download kind executable with dual stack support"
+# We need to replace kind executable and node image 
+# with official release once dual stack is fully supported by upstream.
+curl -L https://github.com/song-jiang/kind/releases/download/dualstack-1.17.0/kind -o ${KIND}
+chmod +x ${KIND}
+exit 0
+
+echo "Create kind cluster"
+${KIND} create cluster --image songtjiang/kindnode-dualstack:1.17.0 --config - <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  disableDefaultCNI: true
+  podSubnet: "192.168.0.0/16,fd00:10:244::/64"
+  ipFamily: DualStack
+nodes:
+# the control plane node
+- role: control-plane
+- role: worker
+- role: worker
+- role: worker
+kubeadmConfigPatches:
+- |
+  apiVersion: kubeadm.k8s.io/v1beta2
+  kind: ClusterConfiguration
+  metadata:
+    name: config
+  featureGates:
+    IPv6DualStack: true
+- |
+  apiVersion: kubeproxy.config.k8s.io/v1alpha1
+  kind: KubeProxyConfiguration
+  metadata:
+    name: config
+  mode: ipvs
+EOF
+
+${kubectl} get no -o wide
+${kubectl} get po --all-namespaces -o wide
+
+echo "Set ipv6 address on each node"
+docker exec kind-control-plane ip -6 a a 2001:20::8/64 dev eth0
+docker exec kind-worker ip -6 a a 2001:20::1/64 dev eth0
+docker exec kind-worker2 ip -6 a a 2001:20::2/64 dev eth0
+docker exec kind-worker3 ip -6 a a 2001:20::3/64 dev eth0
+echo
+
+load_image kind-control-plane
+load_image kind-worker
+load_image kind-worker2
+load_image kind-worker3
+
+echo "Install Calico and Calicoctl for dualstack"
+cp $TEST_DIR/infra/calico-kdd.yaml $TEST_DIR/infra/calico.yaml
+enable_dual_stack $TEST_DIR/infra/calico.yaml
+${kubectl} apply -f $TEST_DIR/infra/calico.yaml
+${kubectl} apply -f $TEST_DIR/infra/calicoctl.yaml
+echo
+
+echo "Wait Calico to be ready..."
+while ! time ${kubectl} wait pod -l k8s-app=calico-node --for=condition=Ready -n kube-system --timeout=300s; do
+    # This happens when no matching resources exist yet,
+    # i.e. immediately after application of the Calico YAML.
+    sleep 5
+done
+echo "Calico is running."
+echo
+
+# Create and monitor a test webserver service for dual stack. 
+echo "Create test-webserver deployment..."
+kubectl apply -f tests/k8st/infra/test-webserver.yaml
```

Comment:
Do we need this for the k8st tests?  If not, I think better to delete this and the associated YAML files.

Reply:
Added connection tests to make sure kind cluster function correctly in terms of dual stack support.